packages/grid/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx

```diff
@@ -38,11 +39,16 @@ const GridVirtualScroller = React.forwardRef<
 >(function GridVirtualScroller(props, ref) {
   const rootProps = useGridRootProps();
   const classes = useUtilityClasses(rootProps);
+  const styles = React.useMemo(
+    () => (rootProps.autoHeight ? {} : { flex: '1 1 0' }),
+    [rootProps.autoHeight],
+  );
 
   return (
     <VirtualScrollerRoot
       ref={ref}
       {...props}
+      style={{ ...props.style, ...styles }}
```

Comment:
Compute the `styles` object once inside `useMemo` instead of destructuring at each render. Otherwise this is a prop change that prevents `React.memo` from working if we use it at some point.